src/util.c

```diff
@@ -923,6 +924,40 @@ sds makePath(char *path, char *filename) {
     return sdscatfmt(sdsempty(), "%s/%s", path, filename);
 }
 
+/* Given the filename, sync the corresponding directory.
+ *
+ * Usually a portable and safe pattern to overwrite existing files would be like:
+ * 1. create a new temp file (on the same file system!)
+ * 2. write data to the temp file
+ * 3. fsync() the temp file
+ * 4. rename the temp file to the appropriate name
+ * 5. fsync() the containing directory */
+int fsyncFileDir(const char *filename) {
+#ifdef _AIX
+    /* AIX is unable to fsync a directory */
+    return 0;
+#endif
+    char temp_filename[PATH_MAX + 1];
+    char *dname;
+    int dir_fd;
+
+    /* In the glibc implementation dirname may modify their argument. */
+    memcpy(temp_filename, filename, strlen(filename) + 1);
+    dname = dirname(temp_filename);
+
+    dir_fd = open(dname, O_RDONLY);
+    if (dir_fd == -1) {
+        return -1;
+    }
```

Comment:
I think we should probably silently ignore `EISDIR` and `EINVAL`.

Reply:
If we will ignore some errno, I think it's better to have a code comment why it's special, but I have no research on these errno...

Reply:
@uvletter Please see https://github.com/redis/redis/pull/10737#discussion_r884124062 . `errno` check will help to detect OSes/Filesystems that do not support fsync() on a directory.